### PR TITLE
[BINS-246] Fixing missing skipped test in XCResult3

### PR DESCRIPTION
**Pull request by viktorbenei**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

xcresult3 converter's output includes the `skipped` tests as `successful` - bug fix

### Changes

- fix of skipped handling
- added unit test for xcresult3's `XML()` func

### CODE REVIEW

File: test/converters/xcresult3/converter_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		filepath.Join("../../testdata/", pathInTestdataDir),
+		dirPathToCopyInto,
+		false,
+	)
+	require.NoError(t, err)
+	return filepath.Join(dirPathToCopyInto, pathInTestdataDir)
+}
+
+func TestConverter_XML(t *testing.T) {
+	t.Run("xcresults3 success-failed-skipped-tests.xcresult", func(t *testing.T) {
```
Comment (godrei) [inline test/converters/xcresult3/converter_test.go:30]: NOTE: what is the reason for running the single test case as a subtest?

**Comment by viktorbenei** [comment]
> is there a way to add sample files to a different repo to prevent bloating of this one? Currently this one is used: https://github.com/bitrise-io/sample-artifacts/tree/master/xcresults

xcresult3 sample moved into the sample-artifacts repo

can you please approva that commit first, and once merged the tests should pass here: https://github.com/bitrise-io/sample-artifacts/pull/20

**Comment by godrei** [comment]
Within the current circumstances, I'm fine with the unit test for `xcresult3` converter.
But in long term, this could lead to issues (like the CI can be run only on macOS). Long term it would be better to mock the actual `xcrun` command execution.

**Comment by BirmacherAkos** [comment]
@viktorbenei Will this solve the following ticket?
https://bitrise.atlassian.net/browse/STEP-1740

**Comment by viktorbenei** [comment]
> @viktorbenei Will this solve the following ticket? https://bitrise.atlassian.net/browse/STEP-1740

Yep @BirmacherAkos I think that ticket is the same as https://bitrise.atlassian.net/browse/BINS-246

File: test/converters/xcresult3/converter_test.go (hunk 1 above)
Comment (viktorbenei) [reply test/converters/xcresult3/converter_test.go:30]: to be able to `defer` the temp directory cleanup

that said, as we only have a single test in it right now it doesn't really matter, but if we'd have another one this way we could generate and cleanup separate temp dirs for them

**Review by godrei — APPROVED** [review]

**Pull request merged**